presto-main/src/main/java/com/facebook/presto/execution/scheduler/SimpleNodeSelector.java

```diff
@@ -99,13 +99,14 @@ public InternalNode selectCurrentNode()
     }
 
     @Override
-    public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks)
+    public SplitPlacementResult computeAssignments(Set<Split> splits, List<RemoteTask> existingTasks, int limit)
     {
         Multimap<InternalNode, Split> assignment = HashMultimap.create();
         NodeMap nodeMap = this.nodeMap.get().get();
         NodeAssignmentStats assignmentStats = new NodeAssignmentStats(nodeTaskMap, nodeMap, existingTasks);
 
-        ResettableRandomizedIterator<InternalNode> randomCandidates = randomizedNodes(nodeMap, includeCoordinator, ImmutableSet.of());
+        ResettableRandomizedIterator<InternalNode> randomCandidates =
```

Comment:
I might be missing something, but I'm not sure i understand how it works.

1. `NodeSelector#computeAssignments` is called multiple times as new splits are scheduled
2. `randomCandidates` is generated every time the `NodeSelector#computeAssignments` is called

Although a single `randomCandidates` size is less then `limit`, it doesn't guarantee same set of nodes to be selected. As `NodeSelector#computeAssignments` is celled again and again - new nodes are likely to be pulled in for scheduling, by effectively eventually returning all the nodes.

Reply:
You are right! I will use a local variable to handle this case.